feat(storage): add set_metadata sample

AlisskaPie opened the pull request.

AlisskaPie committed:
feat(storage): add set_metadata sample

AlisskaPie committed:
subtest in a test

AlisskaPie commented on storage/objects/set_metadata.go:
```diff
@@ -0,0 +1,54 @@
+	o := client.Bucket(bucket).Object(object)
+	objectAttrsToUpdate := storage.ObjectAttrsToUpdate{
+		Metadata: map[string]string{
+			"keyToAddOrUpdate": "value",
```
[Java example](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/SetObjectMetadata.java)

AlisskaPie commented on storage/objects/objects_test.go:
```diff
@@ -173,6 +173,21 @@ func TestObjects(t *testing.T) {
 		t.Errorf("contents = %q; want %q", got, want)
 	}
 
+	t.Run("setMetadata", func(t *testing.T) {
```
subtest because of this comment:
https://github.com/GoogleCloudPlatform/golang-samples/pull/1784#discussion_r510496871

AlisskaPie commented on storage/objects/objects_test.go:
```diff
@@ -173,6 +173,21 @@ func TestObjects(t *testing.T) {
 
+	t.Run("setMetadata", func(t *testing.T) {
+		bkt := client.Bucket(bucket)
+		obj := bkt.Object(object1)
```
bkt, obj because without it we have a problem:
=== RUN   TestObjects/setMetadata
    objects_test.go:187: object content = ""; want "value"

AlisskaPie commented on storage/objects/objects_test.go:
```diff
@@ -173,6 +173,21 @@ func TestObjects(t *testing.T) {
+		if err != nil {
+			t.Errorf("setMetadata: %v", err)
+		}
+		attrs, err := obj.Attrs(ctx)
```
Maybe need to use `getMetadata` to check?
1. But if we'll have a problem with `getMetadata` then 2 tests will fail instead of one.
2. The one subtest for all metadata samples?

IlyaFaer replied:
No, `Attrs` seems okay.

IlyaFaer reviewed (approved):
Seems okay to me

AlisskaPie committed:
Merge branch 'master' into storage_set_metadata

Commit:
docs(securitycenter): wrap samples with future prefix (#1867)

Standardizing Security Command Center samples to use 'securitycenter' prefixing. 

- Wrapped existing samples to keep published doclinks unbroken
- Replace region tags that aren't published. 

Once this PR is through, published sample inclusions will be updated to use the new prefix, then I'll come through again and remove the unused block wraps.

shollyman committed:
chore(bigquery): make flaky sample more verbose in implicating session (#1873)

renovate-bot committed:
chore(all): update all (#1881)

engelke committed:
chore: update GAE samples to Go 1.15 (#1875)

b-loved-dreamer committed:
chore(speech): use GCS file in quickstart (#1878)

munkhuushmgl committed:
chore(videointelligence): increase logo detection timeout to other lang standards (#1885)

katiehochberg committed:
feat: update service directory samples to v1 (#1892)

As our GA launch is approaching, we want to change our samples that populate our documentation to use our v1 api.

tbpg committed:
chore(servicedirectory): fix error text (#1894)

dashpole committed:
chore: update opentelemetry version to v0.15.0 (#1896)

freelerobot committed:
ci(all): allow nightly tests on specific submodules (#1899)

Simple extension of system-test script to allow nightly jobs for specific submodules. 
Job name must have the format: `cloud-devrel/go/golang-samples/system-tests/[OPTIONAL_MODULE_NAME]/[GO_VERSION]`

ahmetb committed:
feat: add h2c sample for Cloud Run (#1887)

ahmetb committed:
fix(run/h2c): instructions (#1913)

Add full command.
Fixes #1912
cc: @tbpg

Commit:
chore(appengine_flexible): rm unused memcache sample (#1907)

jamesward committed:
fix(run/h2c): move http2 to options (#1918)

Commit:
chore(appengine_flexible): rm unused sql samples (#1906)

codyoss committed:
chore: update to pull in latest flakybot (#1917)

tritone committed:
fix(storage): clarify makePublic sample (#1916)

This sample should include the correct values for entity and role
rather than omitting them.

tbpg committed:
test(media): increase number of retries (#1922)

This test gets the job state and checks that it succeeded. The test is
currently regularly failing because the job is still RUNNING and we stop
retrying. Let's double the number of retries and hope that fixes the
issue.

renovate-bot committed:
chore(all): update alpine Docker tag to v3.13 (#1915)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| alpine | final | minor | `3.12` -> `3.13` |

---

### Renovate configuration

:date: **Schedule**: "before 3am on Monday" (UTC).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Renovate will not automatically rebase this PR, because other commits have been found.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/GoogleCloudPlatform/golang-samples).

duluong committed:
fix(endpoints): update gke deployment apiVersion to apps/v1 (#1893)

tbpg committed:
test(monitoring): add retries (#1929)

punya committed:
chore(opentelemetry): update OTel to v0.16.0 (#1935)

No code changes are necessary, this is just to track
that the examples continue to work.

frankyn committed:
Merge branch 'master' into storage_set_metadata

grayside committed:
test(run): increase gcloud retry attempts from 3 to 5 (#1911)

olavloite committed:
test(spanner): unflake backup sample tests (#1932)

Cloud Spanner supports max 1 restore operation at any time on one instance. This can cause problems on test instances if multiple tests are running at the same time. This change adds a retry loop for the specific error that occurs if multiple restore operations are running at the same time.

The same retry loop was also [previously added for Java](https://github.com/googleapis/java-spanner/blob/2c745f31288d5f6ad501b5542ea01db72e015d1e/samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java#L347) where the same problem also sometimes occurred.

Fixes #1900

tbpg committed:
test(media): even more retries (#1937)

The test is still failing because the job is still RUNNING.

Fixes #1880.

freelerobot committed:
docs(logging): feature the correct write structured log sample (#1936)

In this page: https://cloud.google.com/logging/docs/samples/logging-write-log-entry#logging_write_log_entry-go

The go example is mismatched. It should feature the structured-log sample, not the std/idiomatic log sample.

frankyn committed:
Merge branch 'master' into storage_set_metadata

frankyn committed:
Update storage/objects/set_metadata.go

Co-authored-by: Chris Cotter <cjcotter@google.com>
